app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -271,7 +271,9 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
             // if handling new return code (want to return to home screen) but a return at the end of your statement
             switch(requestCode) {
             case PREFERENCES_ACTIVITY:
-                uiController.configUI();
+                // rebuild buttons in case language was changed
+                uiController.setupUI();
```

Comment:
It rebuilds the button data, which is currently only built once, when the adapter is created. Since the button data is only built once, the localized text will get stale if the language changes. So calling setup again here forces the button text to get re-localized.